src/dscanner/analysis/base.d

```diff
@@ -100,3 +103,39 @@ protected:
 
 	MessageSet _messages;
 }
+
+extern(C++) class BaseAnalyzerDmd(AST) : ParseTimeTransitiveVisitor!AST
+{
+	alias visit = ParseTimeTransitiveVisitor!AST.visit;
+
+	extern(D) this(string fileName)
+	{
+		this.fileName = fileName;
+		_messages = new MessageSet;
+	}
+
+	protected string getName()
+	{
+		assert(0);
```

Comment:
What's the purpose of this function?

Reply:
If it has a purpose, add comments in order to document that purpose beyond the comments section of this PR

Reply:
cc @lucica28 this is un-addressed